File: 169-majority-element/169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        Arrays.sort(nums);
        int target = nums.length/2;
        int c = 1;
        for(int i = 1;i<nums.length;i++){
            if(nums[i]==nums[i-1]){
                c++;
            }
            else{
                if(c>target){
                    return nums[i-1];
                }
                else{
                    c=1;
                }
            }
        }
        return nums[nums.length-1];
    }
}

File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        element = 0
        
        # https://www.youtube.com/watch?v=AoX3BPWNnoE&list=PLgUwDviBIf0rPG3Ictpu74YWBQ1CaBkm2&index=17
        for num in nums:
            if count==0:
                element = num
            if num==element:
                count+=1
            else:
                count-=1
        
        return element
        